shell/browser/api/atom_api_url_request.cc

```diff
@@ -259,6 +260,11 @@ void URLRequest::Cancel() {
 }
 
 void URLRequest::Close() {
+  if (!pending_writes_.empty() && last_chunk_written_) {
```

Comment:
How can the pending writes be non-empty but `last_chunk_written_` is true? Perhaps add a comment here?

Reply:
https://github.com/electron/electron/pull/21157/files#diff-83b26000afb8e61c607f454ace5a06d2R328-R329

We  set flags, and let start writing - until ns calls startreading and completes, we have pending bytes ready but flag is set. I also felt this is bit interesting path.